### DISCUSSION TITLE
wire events with pagination is very slow

### DISCUSSION THREAD

**Discussion by apsbd** [livewire/livewire#3007] [7. Bugs (archived)]
### Description

I have a project that has thousands of products in it. In the index page, I made functionality to delete and change featured status using wire:click event listener. But, it takes too long to just load the first page and it takes forever when I change the pages. 

### Exact steps to reproduce

Create an index page with thousands of items. and add a wire:click event listener to each item in the paginated list. It should reproduce the bug

### Stripped-down, copy-pastable code snippets

**Context**
- Livewire version: 2.0
- Laravel version: 8.12

**Comment by richan-fongdasen** [comment]
Hi @apsbd 

Currently, I am working with an index page that displays 100 items per page from a total of 5 thousand items. Every item has four `wire:click` events registered, and there are pagination buttons for navigating through every chunk of data.

As far as i know, I don't see any performance issues at all.

So, can you please tell us more detailed information about the issue, such as:
* How you're performing database queries for the index page
    * Can you confirm that the queries are optimized?
    * Can you confirm there are no N+1 queries?
* How many items per page that you wish to display?
* Are you using the nested Livewire component or blade includes to render the items in a loop?
